advanced-api/automatic-vendor-sharing/app1/webpack.config.js

```diff
@@ -0,0 +1,63 @@
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+const { ModuleFederationPlugin } = require("webpack").container;
+const path = require("path");
+
+// adds all your dependencies as shared modules
+// version is inferred from package.json in the dependencies
+// requiredVersion is used from your package.json
+// dependencies will automatically use the highest available package
+// in the federated app, based on version requirement in package.json
+// multiple different versions might coexist in the federated app
+// Note that this will not affect nested paths like "lodash/pluck"
+// Note that this will disable some optimization on these packages
+// with might lead the bundle size problems
+const deps = require("./package.json").dependencies;
+
+module.exports = {
+  entry: "./src/index",
+  mode: "development",
+  devServer: {
+    contentBase: path.join(__dirname, "dist"),
+    port: 3001,
+  },
+  output: {
+    publicPath: "http://localhost:3001/",
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
+        loader: "babel-loader",
+        exclude: /node_modules/,
+        options: {
+          presets: ["@babel/preset-react"],
+        },
+      },
+    ],
+  },
+  plugins: [
+    new ModuleFederationPlugin({
+      name: "app1",
+      library: { type: "var", name: "app1" },
+      filename: "remoteEntry.js",
+      remotes: {
+        app2: "app2",
+      },
+      exposes: {
+        "./Button": "./src/Button",
+      },
+      shared: {
+        ...deps,
+        react: {
+          singleton: true,
```

Comment:
Is this new to the API now?

Reply:
yep, all things un shares is new to the webpack api. beta.17 is unreleased but will be in a few days